library/core/src/intrinsics/simd.rs

```diff
@@ -0,0 +1,420 @@
+    ///
+    /// `U` must be the element type of `T`.
+    ///
+    /// Starting with the value `y`, multiply the elements of `x` and accumulate.
+    pub fn simd_reduce_mul_ordered<T, U>(x: T, y: U) -> U;
+
+    /// Check if all mask values are true.
+    ///
+    /// `T` must be a vector of integer primitive types.
+    ///
+    /// # Safety
+    /// `x` must contain only `0` or `!0`.
+    pub fn simd_reduce_all<T>(x: T) -> bool;
+
+    /// Check if all mask values are true.
+    ///
+    /// `T` must be a vector of integer primitive types.
+    ///
+    /// # Safety
+    /// `x` must contain only `0` or `!0`.
+    pub fn simd_reduce_any<T>(x: T) -> bool;
+
+    /// Return the maximum element of a vector.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    ///
+    /// For floating-point values, uses IEEE-754 `maxNum`.
+    pub fn simd_reduce_max<T, U>(x: T) -> U;
+
+    /// Return the minimum element of a vector.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    ///
+    /// For floating-point values, uses IEEE-754 `minNum`.
+    pub fn simd_reduce_min<T, U>(x: T) -> U;
+
+    /// Logical "and" all elements together.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    pub fn simd_reduce_and<T, U>(x: T) -> U;
+
+    /// Logical "or" all elements together.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    pub fn simd_reduce_or<T, U>(x: T) -> U;
+
+    /// Logical "exclusive or" all elements together.
+    ///
+    /// `T` must be a vector of integer or floating-point primitive types.
+    ///
+    /// `U` must be the element type of `T`.
+    pub fn simd_reduce_xor<T, U>(x: T) -> U;
+
+    /// Truncate an integer vector to a bitmask.
+    ///
+    /// `T` must be an integer vector.
+    ///
+    /// `U` must be either the smallest unsigned integer with at least as many bits as the length
+    /// of `T`, or the smallest array of `u8` with as many bits as the length of `T`.
+    ///
+    /// Each element is truncated to a single bit and packed into the result.
+    ///
+    /// No matter whether the output is an array or an unsigned integer, it is treated as a single
+    /// contiguous list of bits. The bitmask is always packed on the least-significant side of the
+    /// output, and padded with 0s in the most-significant bits. The order of the bits depends on
+    /// endianess:
+    ///
+    /// * On little endian, the least significant bit corresponds to the first vector element.
+    /// * On big endian, the least significant bit corresponds to the last vector element.
+    ///
+    /// For example, `[-1, 0, -1, -1]` packs to `0b1101` on little endian and `0b1011` on big
```

Comment:
```suggestion
    /// For example, `[!0, 0, !0, !0]` packs to `0b1101` on little endian and `0b1011` on big
```
I think you've been using `!0` and not `-1` elsewhere in this file